pkg/driver_cryptocell_310/psa_cryptocell_310/cipher_chacha20.c

```diff
@@ -163,11 +172,25 @@ psa_status_t psa_cipher_chacha20_finish(psa_cipher_chacha20_ctx_t *ctx,
                                         size_t *output_length)
 {
     DEBUG("Peripheral ChaCha20 Cipher finish.");
-    *output_length = 0;
-    (void)ctx;
-    (void)output;
-    (void)output_size;
-    return PSA_SUCCESS;
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+    if (output_size < ctx->buffer_length)
+    {
```

Comment:
```suggestion
    if (output_size < ctx->buffer_length) {
```

also in other places